cmake/gen-libs/src/main/cpp/gmath/CMakeLists.txt

```diff
@@ -13,14 +13,19 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+
+# Build a static lib and copy it into distribution place for independent app to use
+# we could not delete the bin directory is because android studio would error out
+# when it checks target's existance. Only distribution place is used as import to app
+
 cmake_minimum_required(VERSION 3.4.1)
 
 set(CMAKE_VERBOSE_MAKEFILE on)
 
 add_library(gmath STATIC src/gmath.c)
 
 # copy out the lib binary... need to leave the static lib around to pass gradle check
-set(distribution_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../distribution)
+set(distribution_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../../../../hello-libs/distribution)
 add_custom_command(TARGET gmath POST_BUILD
```

Comment:
You can use set_target_properties to output the library directly to the distribution location, and not have to copy and remove the original. See [here](https://github.com/ggfan/android-ndk/blob/1db95b23c74dd69ec61b6282f7c6bfd51217ef60/hello-libs/libsrc/gperf/CMakeLists.txt).